src/main/java/org/rumbledb/runtime/flwor/expression/SimpleMapExpressionIterator.java

```diff
@@ -59,34 +60,54 @@ public SimpleMapExpressionIterator(
     }
 
     @Override
-    public Item next() {
-        if (this.hasNext) {
-            Item result = this.nextResult; // save the result to be returned
-            setNextResult(); // calculate and store the next result
-            return result;
-        }
-        throw new IteratorFlowException("Invalid next() call in simple map expression", getMetadata());
+    public JavaRDD<Item> getRDDAux(DynamicContext dynamicContext) {
+        JavaRDD<Item> childRDD = this.children.get(0).getRDD(dynamicContext);
+        return childRDD.flatMap((item) -> {
+            List<Item> currentItems = new ArrayList<>();
+            dynamicContext.addVariableValue("$$", currentItems);
+            currentItems.add(item);
+            List<Item> mapValuesRaw = this.rightIterator.materialize(dynamicContext);
+            dynamicContext.removeVariable("$$");
```

Comment:
This might be working by accident, as this concurrently modifies the same dynamicContext object. It would be best, to have full control, to create a separate closure class as it done everywhere else. There, you can create a DynamicContext field in the closure that is chained with the (same, unchanged) original context, and use it instead.

Reply:
I've implemented the change.